### ISSUE TITLE
BUG-000133477 JQuery updates

### ISSUE THREAD

**Issue by aparveen** [Esri/crowdsource-reporter#474]
Many customers are starting to perform static and dynamic analysis of our apps and finding issues related to dated libraries.

for Solutions apps, updating jQuery to 3.5.1 in crowdsource-reporter and crowdsource-manager appears to also require updating bootstrap, bootstrap-datepicker, and moment-with-locales because of changes to functions.

crowdsource-reporter uses

bootstrap 3.3.0 (10/2014)
bootstrap-datepicker 3.0.0 (3/2014)
moment-with-langs 2.7.0 (a predecessor of moment-with-locales that was deprecated 7/2014)
crowdsource-manager uses

bootstrap 3.3.6 (11/2015)
bootstrap-datepicker 4.17.37 (7/2015)
moment-with-locales 2.11.2 (2/2016)
I'm looking at the required call changes to the two apps using the current versions:

jQuery 3.5.1
bootstrap 4.5.2
bootstrap-datepicker 4.17.47
moment-with-locales 2.28.0

**Comment by aparveen** [comment]
I will arrange a meeting with Mike before CT starts to work on it.

**Comment by ashishchoure** [comment]
@MikeTschudi @aparveen 

As per the above comment we updated the following libraries : 
1. jQuery 3.5.1
2. bootstrap 4.5.2
3. bootstrap-datepicker 4.17.47
4. moment-with-locales 2.28.0

However, after updating the libraries we are seeing the following issues in reporter application.  

1. Web Map List : 
   - Web map list UI is getting disturbed.
![image](https://user-images.githubusercontent.com/5525972/94787203-a083ee80-03ef-11eb-9f6a-52f6fb13a8fd.png)
2. Issue List : 
   - The latest bootstrap version has removed the support of glyphicons. Hence the like icon is not being displayed after the like count. (highlighted in yellow color in below screenshot)
![image](https://user-images.githubusercontent.com/5525972/94787444-efca1f00-03ef-11eb-8cad-446378a3e95c.png)
3. Modal Dialog UI (Help and Splash screen) : 
   - There are slight UI distortion in bootstrap modal dialog.
![image](https://user-images.githubusercontent.com/5525972/94787673-3d468c00-03f0-11eb-87b2-e7ffce2f038e.png)
4. Geo form and Comment form :
   -  Date picker icons are missing. (Since glyphicons support is dropped from bootstrap 4)
   -	Issues with Form controls width. 
   -	Success and Error classes are not working. (The support for has-success and has-error classes is removed)
   -	UI issues with cancel and browse button.
   -  After updating date time picker library to the latest version we are seeing following issues:  
       -	Console errors while getting and setting date
       -	Console errors while setting min and max date
       -	Date picker not displayed when date format contains time
       -	Date picker not displayed for range date domains
   -	UI issues in number spinner
   -  The search control UI is being disturbed.
![image](https://user-images.githubusercontent.com/5525972/94788328-091f9b00-03f1-11eb-8d0f-4dac59e82e27.png)
![image](https://user-images.githubusercontent.com/5525972/94788459-3409ef00-03f1-11eb-9ea1-59d0405253e7.png)

5. Details Panel : 
   - Missing icons : Like, Comment and Gallery button. (Since glyphicons support is dropped from bootstrap 4)
   - Minor issues with details panel's width
   - UI Issues with attachments thumbnail inside gallery section
![image](https://user-images.githubusercontent.com/5525972/94789010-05d8df00-03f2-11eb-98cc-531d3902e05c.png)
![image](https://user-images.githubusercontent.com/5525972/94789237-4c2e3e00-03f2-11eb-8be7-802de39177b7.png)

6. Map Search
   - UI issues with Map search
![image](https://user-images.githubusercontent.com/5525972/94789594-b9da6a00-03f2-11eb-9357-eec1768ac00e.png)
 

To summarize, 
- We are seeing issues related to UI and functionalities in almost all the places where we are using bootstrap controls, date time picker and moment library.
-  There are some scenarios where we may have to change the DOM structure in order to resolve the UI issues. However, this may again impact the 508 Accessibility and may require some changes.

Please let us know if you need more details.

**Comment by MikeTschudi** [comment]
If we go ahead with these changes, let's please use the latest versions of all libraries. The versions listed above were simply the latest versions at the time that I looked into updating jQuery.

Regarding `moment-with-locales` in particular, if it can be removed as a part of this work, that would address its upcoming [deprecation](https://momentjs.com/docs/#/-project-status/).

**Comment by ashishchoure** [comment]
### Dev Checklist


Items ​ | Is required​ | Status | Comments​
-- | -- | -- | --
Required NLS   change?​ | No |   | ​
Required   Backward compatibility​? | No |   |  
Required 508​? | Yes |   | 
Need   Sanitize?​ | No |   | ​



### Impact Anaylsis Report



Impacted   Areas | Comments
-- | --
Have   Configuration changes?​ | No
Have Runtime   changes?​ | Yes
Impact on   RTL | Yes
The entire application will be impacted by this change and this will need thorough   testing | Following areas a major impact :
  | Web map list
  | Issue list panel
  | Issue details panel
  | Geo Form and   Comment Form     Functionality
  | Date Picker functionality
  | Number spinner
  | Data Viewer
  | Search Component   (Main map and Geoform map)
  | Help Dialog and   Splash Screen

**Comment by aparveen** [comment]
Verified in QA
